pkg/arvo/sys/hoon.hoon

```diff
@@ -11457,6 +11456,24 @@
     %-  ~(play ut p.vax)
     [%wtgr [%wtts [%leaf %tas -.q.vax] [%& 2]~] [%$ 1]]
   (~(fuse ut p.vax) [%cell %noun %noun])
+::  +shot: deferred +slam
+::
+++  shot
+  |=  [gat=(trap vase) sam=(trap vase)]
+  ^-  (trap vase)
+  =/  [typ=type gen=hoon]
+    :-  [%cell p:$:gat p:$:sam]
+    [%cnsg [%$ ~] [%$ 2] [%$ 3] ~]
+  =+  gun=(~(mint ut typ) %noun gen)
+  |.  ~+
+  [p.gun (slum q:$:gat q:$:sam)]
```

Comment:
This needs a minimized subject outside the `|.` (with `=>  [typ=p.gun +<.$]` or equivalent). To keep that subject maximally minimal, you should also inline the definition of `+slum` inside the `|.`